cask-bootstrap.el

```diff
@@ -47,25 +47,36 @@
 (unless (require 'package nil :noerror)
   (require 'package (expand-file-name "package-legacy" cask-directory)))
 
+(when (version< emacs-version "25.1")
+  ;; Use vendored package-build package (and package-recipe) because its newer
+  ;; versions require Emacs25.1+
+  (require 'package-recipe (expand-file-name "package-recipe-legacy" cask-directory))
+  (require 'package-build (expand-file-name "package-build-legacy" cask-directory)))
+
+
 (let ((orig-load-path load-path))
   (unwind-protect
       (let (package-archives
             package-alist
             package-archive-contents
             (package-user-dir cask-bootstrap-dir))
         (package-initialize)
-        (condition-case nil
-            (mapc 'require cask-bootstrap-packages)
-          (error
-           (add-to-list 'package-archives (cons "gnu" "https://elpa.gnu.org/packages/"))
-           (add-to-list 'package-archives (cons "melpa" "https://melpa.org/packages/"))
-           (package-refresh-contents)
-           (mapc
-            (lambda (package)
-              (unless (package-installed-p package)
-                (package-install package)))
-            cask-bootstrap-packages)
-           (mapc 'require cask-bootstrap-packages))))
+        (message "%S" load-path)
+        (mapc
+         (lambda (package)
+           (condition-case nil
+               (progn
+                 (message "checking package %S" package)
+                 (require package))
+             (error
+              (unless package-archives
+                (add-to-list 'package-archives (cons "gnu" "https://elpa.gnu.org/packages/"))
+                (add-to-list 'package-archives (cons "melpa" "https://stable.melpa.org/packages/"))
+                (package-refresh-contents))
+              (message "installing package %S" package)
+              (package-install package)
+              (require package))))
+         cask-bootstrap-packages))
```

Comment:
Please revert this hunk to clear your change. If you want this change, please create another PR.

Reply:
I am happy to revert the "messages" part, but the workflow where every package is checked and installed one-by-one is important. Previous behaviour was to install all the missing packages if one of the requires fails. Without this change cask will not find some other bootstrap package and still try to install `package-build` along with the others.

Reply:
Ok, let me open a PR with just the change in logic and see if it works better.

Reply:
> I am happy to revert the "messages" part, but the workflow where every package is checked and installed one-by-one is important. Previous behaviour was to install all the missing packages if one of the requires fails. Without this change cask will not find some other bootstrap package and still try to install package-build along with the others.

Nice. It's reasonable.

Reply:
There are separated PR, #493